packages/core/admin/admin/src/utils/uniqueAdminHash.js

```diff
@@ -0,0 +1,24 @@
+export const utils = {
+  bufferToHex(buffer) {
+    return [...new Uint8Array(buffer)].map((b) => b.toString(16).padStart(2, '0')).join('');
+  },
+  async digestMessage(message) {
+    const msgUint8 = new TextEncoder().encode(message);
+    const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);
+
+    return this.bufferToHex(hashBuffer);
+  },
+};
+
+export default async function hashAdminUserEmail(payload) {
+  if (!payload) {
+    return '';
+  }
+  try {
+    return await utils.digestMessage(payload.email);
```

Comment:
In what cases do we need hash.js ?

Reply:
You can follow the discussions above: the browser crypto APIs only work in secure contexts (localhost/ https) and would fail if someone deploys Strapi without using https. If there is a way to get rid of the code-branch I'd be very happy, but I think Ivan had a valid point ...

Reply:
I'm not sure Strapi even works in production deployments if it's not https 🤔  did someone confirmed this ? @ivanThePleasant ?

Reply:
What do you mean by development ? localhost is considered a secure context and will be able to use the crypto already. 

I wasn't asking about do people do it or not but rather can they even do it ? I'm not sure Strapi will work once deployed on a server if it doesn't have https based on our security middleware. Did you try a simple deploy on a VPS without ssl ?

Reply:
I'm pretty sure I've done something like this before on a VPS (not localhost), let me double check and come back on this 👍 Maybe I am confusing something

Reply:
Thanks 🙏

Reply:
Hey @alexandrebodin yes, I've checked it and you're correct, admin panel is not accessible without secure context, so it has to be served over https. So we can get rid of the package then. I don't think there are a lot of people who would try to do something to deliberately serve admin panel over http somehow. And apologies for the delay, I found another bug with this, had to fix it as well

Reply:
Great to hear. That means we can also get rid of the package for the backend part, don't we?

Reply:
I already got rid of it, we use node crypto there now